src/metric/elementwise_metric.cu

```diff
@@ -304,13 +304,17 @@ struct EvalGammaNLogLik {
   }
 
   XGBOOST_DEVICE bst_float EvalRow(bst_float y, bst_float py) const {
-    const bst_float eps = 1e-16f;
-    if (y < eps) y = eps;
-    bst_float psi = 1.0;
+    py = std::max(py, 1e-6f);
+    // hardcoded dispersion.
+    float constexpr kPsi = 1.0;
     bst_float theta = -1. / py;
-    bst_float a = psi;
-    bst_float b = -std::log(-theta);
-    bst_float c = 1. / psi * std::log(y/psi) - std::log(y) - common::LogGamma(1. / psi);
+    bst_float a = kPsi;
+    // b = -std::log(-theta);
+    float b = 1.0f;
+    // c = 1. / kPsi * std::log(y/kPsi) - std::log(y) - common::LogGamma(1. / kPsi);
+    //     1.0f      * 1.0f             - 1.0f        - 0
```

Comment:
If `y` is a positive value, `std::log(y)` is not necessarily equal to 1.0f, I think.

Reply:
There are 2 log y, they cancel out.

Reply:
Thanks, I will suggest a small change to clarify this point.

Reply:
The 1.0 is misleading, will remove it.